src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -97,8 +108,28 @@ ol.renderer.canvas.VectorLayer.prototype.composeFrame =
     var alpha = replayContext.globalAlpha;
     replayContext.globalAlpha = layerState.opacity;
     replayGroup.replay(
-        replayContext, frameState.pixelRatio, transform,
-        frameState.viewState.rotation, frameState.skippedFeatureUids);
+        replayContext, pixelRatio, transform, rotation, skippedFeatureUids);
+
+    if (vectorSource.getWrapX() && projection.isGlobal() &&
+        !ol.extent.containsExtent(projectionExtent, frameState.extent)) {
+      var startX = extent[0];
+      var worldWidth = ol.extent.getWidth(projectionExtent);
+      var world = 0;
+      while (startX < projectionExtent[0]) {
+        transform = this.getTransform(frameState, worldWidth * (--world));
+        replayGroup.replay(
+            replayContext, pixelRatio, transform, rotation, skippedFeatureUids);
+        startX += worldWidth;
+      }
+      world = 0;
+      startX = extent[2];
+      while (startX > projectionExtent[2]) {
+        transform = this.getTransform(frameState, worldWidth * (++world));
```

Comment:
As above.  Personal preference, but I like to only use the increment or decrement operators when they are the only operator in the statement.